### PR TITLE
TST: test_kolmogorov xfail 32-bit

### PR DESCRIPTION
**Pull request by tylerjereddy**
Another patch aimed at fixing 32-bit wheel build issues described in https://github.com/MacPython/scipy-wheels/pull/37 & ~with local verification of fix using an i386 docker container as described in #9493.~

~`test_oneovernclose()` seemed to need two "fixes" to pass:~

~1. use a sized (`np.float32`) type since `float` is apparently a C double (`float64`) in an i386 Python interpreter on 64-bit architecture~
~2. reduce the tolerance stringency to something more suitable to single precision~

~That said, the above fix is only necessary for one of the two tests of this type & even better may be to robustly & portably detect interpreter architecture & dispatch the test stringency from there, maybe. Not sure if that is in scope for this fix / for now.~

### CODE REVIEW

**Comment by WarrenWeckesser** [comment]
@tylerjereddy wrote:

> 1. use a sized (np.float32) type since float is apparently a C double (float64) in an i386 Python interpreter on 64-bit architecture

I am confused by this statement.  Are there any platforms or architectures out there where a Python `float` is *not* a IEEE 754 64 bit floating point number?   (I haven't looked into the problem you are trying to solve, so your fix might be OK, but the justification sounds fishy.)

**Comment by rgommers** [comment]
> Are there any platforms or architectures out there where a Python float is not an IEEE 754 64 bit floating point number? 

should always be the case AFAIK

File: scipy/special/tests/test_kolmogorov.py [hunk 1]
```diff
@@ -237,10 +237,10 @@ def test_oneovernclose(self):
         dataset0 = np.column_stack([n, x, pp])
         FuncData(_smirnovp, dataset0, (0, 1), 2, rtol=_rtol).check(dtypes=[int, float, float])
 
-        x = 1.0/n + 2*np.finfo(float).eps
+        x = 1.0/n + 2*np.finfo(np.float32).eps
```
Comment (rgommers) [inline scipy/special/tests/test_kolmogorov.py]: does the test really fail without this line but with the 1e-5 rtol? it's probably the `int` that's the problem, that'll be int32 on 32-bit platform - `float` on the other hand is always float64.

**Comment by rgommers** [comment]
I'd prefer to mark this test as knownfail in 1.2.x, and then investigate what's really going on here and fix it correctly later on in master.

File: scipy/special/tests/test_kolmogorov.py (hunk 1 above)
Comment (tylerjereddy) [reply scipy/special/tests/test_kolmogorov.py]: Yes, it really fails in that scenario; I tried a bunch of permutations on the types / tolerances to produce a minimally invasive fix.

**Comment by tylerjereddy** [comment]
> I am confused by this statement. Are there any platforms or architectures out there where a Python float is not an IEEE 754 64 bit floating point number? (I haven't looked into the problem you are trying to solve, so your fix might be OK, but the justification sounds fishy.)

Well if the 64-bit floating point number is involved in **any** kind of casting or comparison operation internally on a 32-bit interpreter it is certainly possible that an issue may arise that doesn't if it were typed 32-bit, but I'll go ahead and just adjust the unit test to skip or xfail as requested.

**Comment by charris** [comment]
AFAIK, Python "float" is always double precision. The C conventions, "float", "double", "long double" are different and not to be confused with the Python convention.

**Comment by WarrenWeckesser** [comment]
@charris wrote:
> The C conventions, "float", "double", "long double" are different 

Is there any compiler in wide use today that you know of in which `float` is not 32 bit (single precision) or `double` is not 64 bit?

**Comment by tylerjereddy** [comment]
Python `float` is always double precision and that's the problem here -- we don't understand why without dissecting the internal code a bit more so I'll apply the xfail type flag instead. Is it really hard to believe that a high-precision internal comparison or a casting operation somewhere might break on 32-bit interpreter but not 64-bit when the input is higher precision?

**Comment by tylerjereddy** [comment]
Revised based on comments from @rgommers above -- the conditional `xfail` is based on the 64-bit platform `IS_64BIT` check used in the NumPy test suite in a few places.

**Comment by tylerjereddy** [comment]
I didn't check for the test skip locally in an i386 docker container again though -- perhaps that's something that belongs in main repo CI eventually.

**Pull request merged**